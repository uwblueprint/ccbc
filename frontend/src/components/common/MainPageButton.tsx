import React from "react";
import { useHistory } from "react-router-dom";
import { HOME_PAGE } from "../../constants/Routes";

const MainPageButton = (): React.ReactElement => {
  const history = useHistory();
  const navigateTo = () => history.push(HOME_PAGE);
  return (
    <div>
      <button
        onClick={navigateTo}
        className="btn btn-primary"
        type="button"
        style={{ textAlign: "center" }}
      >
        Go Back
      </button>
    </div>
  );
};

export default MainPageButton;
