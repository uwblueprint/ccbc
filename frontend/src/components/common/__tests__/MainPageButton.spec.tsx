import React from "react";
import { create } from "react-test-renderer";

import MainPageButton from "../MainPageButton";

// Example snapshot test.
// For more information on React testing, visit:
// https://jestjs.io/docs/tutorial-react
// https://reactjs.org/docs/testing.html

describe("MainPageButton component", () => {
  it("should match the snapshot", () => {
    const button = create(<MainPageButton />);
    expect(button.toJSON()).toMatchSnapshot();
  });
});
