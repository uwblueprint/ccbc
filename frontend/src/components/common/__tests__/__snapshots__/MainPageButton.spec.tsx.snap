// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPageButton component should match the snapshot 1`] = `
<div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    style={
      Object {
        "textAlign": "center",
      }
    }
    type="button"
  >
    Go Back
  </button>
</div>
`;
